src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/AuditService.kt

```diff
@@ -0,0 +1,158 @@
+package uk.gov.justice.digital.hmpps.pecs.jpc.service
+
+import com.beust.klaxon.Klaxon
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.pecs.jpc.auditing.AuditEvent
+import uk.gov.justice.digital.hmpps.pecs.jpc.auditing.AuditEventRepository
+import uk.gov.justice.digital.hmpps.pecs.jpc.auditing.AuditEventType
+import uk.gov.justice.digital.hmpps.pecs.jpc.auditing.AuditableEvent
+import uk.gov.justice.digital.hmpps.pecs.jpc.config.TimeSource
+import uk.gov.justice.digital.hmpps.pecs.jpc.location.LocationType
+import uk.gov.justice.digital.hmpps.pecs.jpc.price.Money
+
+@Service
+@Transactional
+class AuditService(private val auditEventRepository: AuditEventRepository) {
+  @Autowired
+  private lateinit var timeSource: TimeSource
+
+  fun createLogInEvent(username: String) {
```

Comment:
I know this was my doing from earlier but we could probably go a bit further and use a factory pattern for creation of the auditable events.  This could simplify/reduce the number of methods down in the service e.g.

```
class AuditServcie(.....) {

  fun createEvent(event:  AuditableEvent) {
    // code blurb goes here...
  }
}

```